src/WebJobs.Script/Rpc/Configuration/WorkerConfigFactory.cs

```diff
@@ -220,5 +239,83 @@ internal string GetExecutablePathForJava(string defaultExecutablePath)
                 return Path.GetFullPath(Path.Combine(javaHome, "bin", defaultExecutablePath));
             }
         }
+
+        internal bool IsHydrationNeeded(string workerPath)
+        {
+            if (string.IsNullOrEmpty(workerPath))
+            {
+                return false;
+            }
+
+            return workerPath.Contains(LanguageWorkerConstants.OSPlaceholder) ||
+                    workerPath.Contains(LanguageWorkerConstants.ArchitecturePlaceholder) ||
+                    workerPath.Contains(LanguageWorkerConstants.RuntimeVersionPlaceholder);
+        }
+
+        internal string GetHydratedWorkerPath(WorkerDescription description)
+        {
+            string workerPath = description.GetWorkerPath();
+            if (string.IsNullOrEmpty(workerPath))
+            {
+                return null;
+            }
+
+            OSPlatform os = _systemRuntimeInformation.GetOSPlatform();
```

Comment:
handle case if os is null

Reply:
OSPlatform is a struct and cannot be `null` so we should be good there (https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/structs)